### PR TITLE
Add main view + menu (+ JL)

### PR DESCRIPTION
**Pull request by samdoiron**

### CODE REVIEW

File: app/src/test/java/com/umbrella/umbrella/MainPresenterTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd");
+            Date expectedDate = format.parse("2018-12-25");
+            FakeRegistrationInfoRepo fakeRepo = new FakeRegistrationInfoRepo(expectedDate);
+            MainPresenter presenter = new MainPresenter(fakeRepo);
+
+            MainViewModel viewModel = presenter.getViewModel();
+
+           assertEquals("Registration ends on December 25 2018", viewModel.deadlineMessage);
+        } catch (ParseException e) {
+            fail("Failed ot parse test date");
```
Comment (wauchoda) [inline app/src/test/java/com/umbrella/umbrella/MainPresenterTest.java]: Misspelling "Failed to parse test date"

File: app/src/test/java/com/umbrella/umbrella/MainPresenterTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd");
+            Date expectedDate = format.parse("2018-12-01");
+            FakeRegistrationInfoRepo fakeRepo = new FakeRegistrationInfoRepo(expectedDate);
+            MainPresenter presenter = new MainPresenter(fakeRepo);
+
+            MainViewModel viewModel = presenter.getViewModel();
+
+            assertEquals("Registration ends on December 1 2018", viewModel.deadlineMessage);
+        } catch (ParseException e) {
+            fail("Failed ot parse test date");
```
Comment (wauchoda) [inline app/src/test/java/com/umbrella/umbrella/MainPresenterTest.java]: Misspelling "Failed to parse test date"

File: app/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -1,11 +1,17 @@
 <resources>
```
Comment (wauchoda) [inline app/src/main/res/values/styles.xml]: Remove whitespace

File: app/src/main/res/menu/drawer_view.xml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android" >
+    <group android:checkableBehavior="single">
+        <item
+            android:id="@+id/nav_camera"
+            android:title="View schedule" />
```
Comment (wauchoda) [inline app/src/main/res/menu/drawer_view.xml]: Inconsistent capitalization between navigation items (capitalize the first letter of each word)

File: app/src/main/res/menu/drawer_view.xml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+        <item
+            android:id="@+id/nav_add_drop"
+            android:title="Add / drop courses" />
```
Comment (wauchoda) [inline app/src/main/res/menu/drawer_view.xml]: Inconsistent capitalization between navigation items (capitalize the first letter of each word)

File: app/src/main/java/com/umbrella/umbrella/MainPresenter.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    /** @return the view model for the main screen */
+    public MainViewModel getViewModel() {
+        MainViewModel viewModel = new MainViewModel();
+        viewModel.deadlineMessage = getDeadlineMessage();
+        return viewModel;
+    }
+
+    private String getDeadlineMessage() {
```
Comment (wauchoda) [inline app/src/main/java/com/umbrella/umbrella/MainPresenter.java]: java doc

File: app/src/main/java/com/umbrella/umbrella/FakeRegistrationInfoRepo.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+ * Created by samdoiron on 2018-03-07.
+ */
+
+public class FakeRegistrationInfoRepo implements RegistrationInfoRepo {
+    private final Date deadline;
+
+    public FakeRegistrationInfoRepo(Date deadline) {
+        this.deadline = deadline;
+    }
+    public Date getRegistrationDeadline() {
```
Comment (wauchoda) [inline app/src/main/java/com/umbrella/umbrella/FakeRegistrationInfoRepo.java]: javadoc

File: app/src/main/java/com/umbrella/umbrella/MainPresenter.java (hunk 6 above)
Comment (samdoiron) [reply app/src/main/java/com/umbrella/umbrella/MainPresenter.java]: Private API

File: app/src/main/res/menu/drawer_view.xml (hunk 5 above)
Comment (samdoiron) [reply app/src/main/res/menu/drawer_view.xml]: Made others consistent

File: app/src/main/res/menu/drawer_view.xml (hunk 4 above)
Comment (samdoiron) [reply app/src/main/res/menu/drawer_view.xml]: Made others consistent

File: app/src/main/java/com/umbrella/umbrella/FakeRegistrationInfoRepo.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+ */
+
+public class FakeRegistrationInfoRepo implements RegistrationInfoRepo {
+    private final Date deadline;
+
+    public FakeRegistrationInfoRepo(Date deadline) {
+        this.deadline = deadline;
+    }
+
+    /** Return the date when the registration period next ends */
```
Comment (jsedge) [inline app/src/main/java/com/umbrella/umbrella/FakeRegistrationInfoRepo.java]: JavaDoc

**Comment by samdoiron** [comment]
🚂  MERGE TRAIN COMING IN 🚂

**Pull request merged**